xios_examples/read_unstructured_domain_resample/test_resample_cases.py

```diff
@@ -0,0 +1,55 @@
+import copy
+import glob
+import netCDF4
+import numpy as np
+import os
+import subprocess
+import unittest
+
+import xios_examples.shared_testing as xshared
+
+this_path = os.path.realpath(__file__)
+this_dir = os.path.dirname(this_path)
+
+class TestResampleDomain(xshared._TestCase):
+    test_dir = this_dir
+    transient_inputs = ['domain_input.nc', 'domain_input_ugrid.nc']
+    transient_outputs = ['domain_output.nc', 'domain_output_ugrid.nc']
+    rtol = 5e-03
+    mesh_file = 'mesh_C12.nc'
+
+
+# A list of input `.cdl` files where XIOS is known to produce different
+# output from the expected output data
+# for future investigation / ToDo
+# this is a dict, where the name of the key is the name of the test
+# to register as a known_failure (tname)
+# and the value is a string explaining the failure
+# this handles FAIL conditions but NOT ERROR conditions
+known_failures = {'test_sinusiod':
+                  ('The difference file has large errors'),
+                  'test_harmonic':
+                  ('The difference file has large errors'),
+                  'test_vortex':
+                  ('The difference file has large errors'),
+                  'test_gulfstream':
+                  ('The difference file has large errors')
+                  }
+
+# iterate over analytic function names
+for f in ['sinusiod', 'harmonic', 'vortex', 'gulfstream']:
+    # unique name for the test
+    tname = f'test_{f}'
+    # add the test as an attribute (function) to the test class
+    if os.environ.get('MVER', '').startswith('XIOS3/trunk'):
+        # these tests are hitting exceptions with XIOS3
+        # but not XIOS2, so skip for XIOS3 runner
+        setattr(TestResampleDomain, tname,
+                unittest.skip(TestResampleDomain.make_a_resample_test(f)))
```

Comment:
please may i check, are there exceptions with XIOS3 for these tests, different from the `known_failures`?
if so, what is XIOS3 reporting?

Reply:
Just had a look at the documentation and there is a `read_access attribute` but it is for a field not a file as I originally tried. From the user guide

> In order not to waste memory, the instant data of a field can be read from the
> model only if:
> • it is part of a file whose attribute mode is “read”
> • or its attribute read_access is set to “true”.
> In any other case, trying to access the field data would cause an error.

Looks like this has changed in XIOS3

After adding the `read_access` attribute I no longer got the above error I now get this error
```
Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x7f38a25483ff in ???
#1  0x7f38a61a4caa in _ZNKSs4_Rep12_M_is_leakedEv
        at /home/conda/feedstock_root/build_artifacts/gcc_compilers_1715013693798/work/libstdc++-v3/include/bits/cow_string.h:214
#2  0x7f38a61a5da7 in _ZNSs4_Rep7_M_grabERKSaIcES2_
        at /home/conda/feedstock_root/build_artifacts/gcc_compilers_1715013693798/work/libstdc++-v3/include/bits/cow_string.h:265
#3  0x7f38a61a5dee in _ZNSsC2ERKSs
        at /home/conda/feedstock_root/build_artifacts/gcc_compilers_1715013693798/work/libstdc++-v3/include/bits/cow_string.h:546
#4  0x55c4acc707c0 in _ZN4xios20CTransformationPaths23getNextTransformationIdEv
        at /home/users/jcole/software/xios3_trunk_r2629/src/transformation/transformation_path.cpp:97
#5  0x55c4ac99c806 in _ZN4xios5CGrid24buildTransformationGraphERNS_17CGarbageCollectorEbPS0_ddRS3_bPNS_6CFieldE
        at /home/users/jcole/software/xios3_trunk_r2629/src/node/grid.cpp:2141
#6  0x55c4ac91ef0c in _ZN4xios6CField18buildWorkflowGraphERNS_17CGarbageCollectorE
        at /home/users/jcole/software/xios3_trunk_r2629/src/node/field.cpp:746
#7  0x55c4ac652500 in _ZN4xios8CContext15closeDefinitionEv
        at /home/users/jcole/software/xios3_trunk_r2629/src/node/context.cpp:1208
#8  0x55c4acb28e41 in cxios_context_close_definition
        at /home/users/jcole/software/xios3_trunk_r2629/src/interface/c/icdata.cpp:133
#9  0x55c4ac2bfe37 in __idata_MOD_xios_close_context_definition
        at /home/users/jcole/software/xios3_trunk_r2629/config_jasmin_debug/ppsrc/xios/interface/fortran/idata.f90:691
#10  0x55c4ac26f7cd in initialise
        at /home/users/jcole/software/git/jeff-cole/tcd-XIOS-demonstration/xios_examples/read_unstructured_domain_resample/resample.F90:62
#11  0x55c4ac26f45c in resample
        at /home/users/jcole/software/git/jeff-cole/tcd-XIOS-demonstration/xios_examples/read_unstructured_domain_resample/resample.F90:18
#12  0x55c4ac26fd98 in main
        at /home/users/jcole/software/git/jeff-cole/tcd-XIOS-demonstration/xios_examples/read_unstructured_domain_resample/resample.F90:9

```

Is this similat to what you saw for the other tests?

Reply:
hi @jeff-cole 

Many of the tests run fine with XIOS2 & XIOS3 (there are some exceptions, but we'd like to keep these to a minimum if we can, and explore to try and have version-portable example code)

Where & how did you run these tests to get the `invalid_memory_reference` error please?

i've just tried branching from your branch and enabling these test for XIOS3 in a fork-branch:
https://github.com/mo-marqh/tcd-XIOS-demonstration/tree/read_unstructured_domain_resample
https://github.com/mo-marqh/tcd-XIOS-demonstration/commit/7640c1ed5014a08976469b8bcb64038226e05b47
I have set off the github runners within my fork seems to run these tests for XIOS3 and complete, with the same tolerance known-error, but no Segmentation fault 
https://github.com/mo-marqh/tcd-XIOS-demonstration/actions/runs/9157120279/job/25172852674#step:6:92

it would be good to try and understand this environment difference if we can.
In the meantime, please may you remove these skip patterns for these tests and run them on github for this branch?

thank you
mark

Reply:
Hi @mo-marqh 

I think using exceptedFailure hides errors other than not meeting the tolerance requirements. When I turned off skipping the XIOS3 test it passed with 4 known failures. When I then increased the tolerance value and removed the known failures it failed with a python netCDF4 problem, which was previously masked by using exceptedFailure.

I fixed the netCDF4 problem (I was using a feature which was added after the version being using by the github runners) and now it fails with the same error as I got above (not using `read_access`). I will fix this and see if I get the `invalid_memory_reference` error.

Reply:
I didn't get the same `invalid_memory_reference` error, this time I got
```
/home/runner/work/tcd-XIOS-demonstration/tcd-XIOS-demonstration/xios_examples/read_unstructured_domain_resample/xios_client_0.err
-> error : WARNING: Unexpected request for buffer to communicate with server 0
In file "type_impl.hpp", function "void xios::CType<T>::_checkEmpty() const [with T = int]",  line 210 -> Data is not initialized
```
I also tried running it on archer2 (not with the trunk but 2 versions earlier) and this seemed to get  further but hung.

All of the read_axis/read_domain tests skip running XIOS3, can you remember what problem they had which made you skip them?

Reply:
My original runs were done on jasmin, this is what gave the `invalid_memory_reference` error,